Source/WebCore/css/StyleSheetContents.cpp

```diff
@@ -532,18 +532,22 @@ bool StyleSheetContents::traverseRules(NOESCAPE const Function<bool(const StyleR
     return traverseRulesInVector(m_childRules, handler);
 }
 
-bool StyleSheetContents::hasNestingRules() const
+bool StyleSheetContents::hasLateResolvedRules() const
 {
-    if (m_hasNestingRulesCache)
-        return *m_hasNestingRulesCache;
+    if (m_hasLateResolvedRulesCache)
+        return *m_hasLateResolvedRulesCache;
 
-    m_hasNestingRulesCache = traverseRulesInVector(m_childRules, [&] (auto& rule) {
+    m_hasLateResolvedRulesCache = traverseRulesInVector(m_childRules, [&] (const auto& rule) {
         if (rule.isStyleRuleWithNesting())
             return true;
+        if (rule.isScopeRule())
+            return true;
+        if (rule.isNestedDeclarationsRule())
+            return true;
```

Comment:
I think the existing `rule.isStyleRuleWithNesting()` test is sufficient as these other cases are only possible/problematic when it is present.